src/main/java/org/myrobotlab/service/BodyPart.java

```diff
@@ -219,4 +226,108 @@ static public ServiceType getMetaData() {
     meta.addDependency("org.apache.commons", "commons-lang3", "3.3.2");
     return meta;
   }
+
+  //pasted from inmoov service, for compatibility
+  @Override
+  public void onJointAngles(Map<String, Double> angleMap) {

```

Comment:
This method is implemented specifically for the InMoovArm.   It does not generically apply to any "BodyPart".  It should not be copy and pasted here.  Implementing IKJointAngleListener is fine.. but rather than assuming it's an inmoov arm.. log a warning that it's not implemented yet and go on.  (or throw a NotImplementedExeception if you wish.)